File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Generecidad/Ejercicio3/Java/Producto.java
package Generecidad.Ejercicio3.Java;

public class Producto implements Buscarable {
    private String nombre;
    private String codigo;
    private double precio;

    public Producto(String nombre, String codigo, double precio) {
        this.nombre = nombre;
        this.codigo = codigo;
        this.precio = precio;
    }

    public String getNombre() {
        return nombre;
    }

    public String getCodigo() {
        return codigo;
    }

    @Override
    public String toString() {
        return String.format("Producto: '%s' (Codigo: %s, Precio: $%.2f)", nombre, codigo, precio);
    }

    @Override
    public boolean esSimilar(String criterio) {
        String lowerCaseCriterio = criterio.toLowerCase();
        return this.nombre.toLowerCase().contains(lowerCaseCriterio) ||
               this.codigo.toLowerCase().contains(lowerCaseCriterio);
    }
}

File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Generecidad/Ejercicio3/Python/Producto.py
class Producto:
   
    def __init__(self, nombre: str, codigo: str, precio: float):
        self.nombre = nombre
        self.codigo = codigo
        self.precio = precio

    def __str__(self):
        return f"Producto: '{self.nombre}' (Codigo: {self.codigo}, Precio: ${self.precio:.2f})"

    def __repr__(self):
        return self.__str__()

    def es_similar(self, criterio: str) -> bool:
        return criterio.lower() in self.nombre.lower() or criterio.lower() in self.codigo.lower()
